langserver/handlers/execute_command.go

```diff
@@ -7,26 +7,35 @@ import (
 	"strings"
 
 	"github.com/creachadair/jrpc2/code"
+	lsctx "github.com/hashicorp/terraform-ls/internal/context"
 	lsp "github.com/sourcegraph/go-lsp"
 )
 
+const commandPrefix = "terraform-ls."
+
 type executeCommandHandler func(context.Context, commandArgs) (interface{}, error)
 type executeCommandHandlers map[string]executeCommandHandler
 
 var handlers = executeCommandHandlers{
 	"rootmodules": executeCommandRootModulesHandler,
 }
 
-func (h executeCommandHandlers) Names() []string {
-	var names []string
+func (h executeCommandHandlers) Names(suffix string) []string {
+	if suffix != "" && !strings.HasPrefix(suffix, ".") {
+		suffix = "." + suffix
+	}
+	var fullnames []string
 	for name := range h {
-		names = append(names, name)
+		fullnames = append(fullnames, commandPrefix+name+suffix)
 	}
-	return names
+	return fullnames
 }
 
 func (lh *logHandler) WorkspaceExecuteCommand(ctx context.Context, params lsp.ExecuteCommandParams) (interface{}, error) {
-	handler, ok := handlers[params.Command]
+	serverID, _ := lsctx.ServerID(ctx)
+	name := strings.TrimPrefix(params.Command, commandPrefix)
+	name = strings.TrimSuffix(name, "."+serverID)
+	handler, ok := handlers[name]
```

Comment:
Could we perhaps simplify this logic here by decoupling the lookup into a dedicated `executeCommandHandlers` method - say `FindCommand(name, suffix string) (executeCommandHandler, bool)` - then we can also more easily utilize `Names()` there and perform exact match on the key, instead of string key trimming?

Reply:
I can isolate this code a bit more certainly. I found it easier to actually keep the prefix and suffix out of the map, so I likely will still need the trimming but I'll see what I can do